### PR TITLE
Provision blob storage

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Provision new BLOB](https://tools.hmcts.net/jira/browse/BPS-805)

### Change description ###

Same applied in hmcts/blob-router-service#8

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: blob-storage.tf [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+locals {
+  stripped_product  = "${replace("${var.product}", "-", "")}"
+  account_name      = "${local.stripped_product}${var.env}"
+  mgmt_network_name = "${var.subscription == "prod" || var.subscription == "nonprod" ? "mgmt-infra-prod" : "mgmt-infra-sandbox"}"
+  external_hostname = "${local.stripped_product}${var.env == "prod" ? "" : join("", [".", var.env])}.${var.external_hostname}"
+
+  // for each client service two containers are created: one named after the service
+  // and another one, named {service_name}-rejected, for storing envelopes rejected by process
+  client_containers = ["bulkscan", "cmc", "crime", "divorce", "finrem", "probate", "sscs"]
```
Comment (mdayican) [inline blob-storage.tf]: do we need all ?

File: blob-storage.tf [hunk 2]
```diff
@@ -0,0 +1,83 @@
+provider "azurerm" {
```
Comment (timja) [inline blob-storage.tf]: any reason you're not including the app gateway config?

File: blob-storage.tf (hunk 1 above)
Comment (doncem) [reply blob-storage.tf]: yes. this should reflect existing + crime

File: blob-storage.tf (hunk 2 above)
Comment (doncem) [reply blob-storage.tf]: wasn't aware i'll have to include palo in here :). was wondering whether app gateway should go in this pr or together with https://tools.hmcts.net/jira/browse/BPS-870 ticket

Comment (timja) [reply blob-storage.tf]: you can't test it without the app gateway, whether this PR or another, it's up to you.

Comment (doncem) [reply blob-storage.tf]: fair enough. i'll include here - all in one :)

**Pull request merged**